solvation_analysis/plotting.py

```diff
@@ -0,0 +1,319 @@
+                raise Exception("Invalid value of keep_solvents. \n keep_solvents: " +
+                                str(keep_solvents) + "\n Valid options for keep_solvents: " +
+                                str(list(properties[solution].keys()))) from None
+
+    # generate figure and make a DataFrame of the data
+    fig = go.Figure()
+    df = pd.DataFrame(data=properties.values())
+    df.index = list(properties.keys())
+
+    if series and x_axis == "species":
+        # each solution is a line
+        df = df.transpose()
+        fig = px.line(df, x=df.index, y=df.columns, labels={"variable": legend_label})
+        fig.update_xaxes(type="category")
+    elif series and x_axis == "solution":
+        # each species is a line
+        fig = px.line(df, x=df.index, y=df.columns, labels={"variable": legend_label})
+        fig.update_xaxes(type="category")
+    elif not series and x_axis == "species":
+        # each solution is a bar
+        df = df.transpose()
+        fig = px.bar(df, x=df.index, y=df.columns, barmode="group", labels={"variable": legend_label})
+    elif not series and x_axis == "solution":
+        # each species is a bar
+        fig = px.bar(df, x=df.index, y=df.columns, barmode="group", labels={"variable": legend_label})
+    return fig
+
+
+def compare_free_solvents(solutions):
+    # this should be a grouped vertical bar chart or a line chart
+    # 1.0 should be marked and annotated with a dotted line
+    fig = compare_solvent_dicts()
+    return
+
+
+def compare_pairing(solutions, coerce_solvent_names=None, keep_solvents=None, x_label="Solvent", y_label="Pairing", title="Graph of Pairing Data", legend_label="Legend", **kwargs):
+    # this should be a grouped vertical bar chart or a line chart
+    # 1.0 should be marked and annotated with a dotted line
+    """
+    Compares the pairing of multiple solutions.
+    Parameters
+    ----------
+    solutions : a dictionary of Solution objects
+    coerce_solvent_names : a dictionary where the keys are strings of solvent names and the values are
+        strings of a more generic name for the solvent (i.e. {"EAf" : "EAx", "fEAf" : "EAx"})
+    keep_solvents : a list of strings of solvent names that are common to all systems in question,
+        graphed in the order that is passed into the function
+    x_label : name of the x-axis as a string
+    y_label : name of the y-axis as a string
+    title : title of figure as a string
+    legend_label : title of legend as a string
+    kwargs : consists of the x_axis and series parameters
+        x_axis : a string specifying "species" or "solution" to be graphed on the x_axis
+        series : Boolean (False for a bar graph; True for a line graph)
+
+    Returns
+    -------
+    fig : Plotly.Figure
+
+    """
+    coerce_solvent_names = coerce_solvent_names or {}
+    pairing = {solution : solutions[solution].pairing.pairing_dict for solution in solutions}
```

Comment:
Let's move away calling anything a `solution` if it isn't a `Solution` object. If it's a `str` solution name, we should call it `solution_name`. Well chosen variable names make it far easier to understand the code!

There are a few other places where this needs to be fixed too.